JavaScript/JavaScriptSDK/ajax/ajax.ts

```diff
@@ -132,7 +134,11 @@ module Microsoft.ApplicationInsights {
         }
 
         private sendHandler(xhr: XMLHttpRequestInstrumented, content) {
-            xhr.ajaxData.requestSentTime = dateTime.Now();            
+            xhr.ajaxData.requestSentTime = dateTime.Now();
+            if (!this.appInsights.config.disableCorrelationHeaders && (UrlHelper.parseUrl(xhr.ajaxData.getAbsoluteUrl()).host == this.currentWindowHost)) {
+                xhr.setRequestHeader("x-ms-request-root-id", this.appInsights.context.operation.id);
+                xhr.setRequestHeader("x-ms-request-id", xhr.ajaxData.id);
+            }
```

Comment:
Should we have try...catch...log around the whole setting headers part to ensure we don't break ajax with this functionality?